.github/workflows/maven-build-publish.yml

```
@@ -0,0 +1,52 @@
+name: Maven CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  release:
+    types: [published]
+
+jobs:
+  validate:
+    uses: ozone-his/.github/.github/workflows/maven.yml@main
+    with:
+      java-version: 8
+    secrets: inherit
+  
+  spotless:
+    needs: validate
+    uses: ozone-his/.github/.github/workflows/check-spotless.yml@main
+    with:
+      java-version: 8
+
+  # This job publishes snapshots on every push to main
+  pre-release:
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    needs: validate
```

Mostly because we shouldn't be publishing anything unless the spotless check passes. Without this change, its valid for the package to be published even if the spotless check fails.

```suggestion
    needs: spotless
```